test/python/qpy/test_circuit_load_from_qpy.py

```diff
@@ -138,6 +138,45 @@ def test_empty_layout(self):
         qc._layout = TranspileLayout(None, None, None)
         self.assert_roundtrip_equal(qc)
 
+    def test_overlapping_definitions(self):
```

Comment:
Wouldn't it make more sense to have this test be part of `TestLoadFromQPY`(in `test/python/circuit/test_circuit_load_from_qpy.py`) rather than `TestLayout`? Could this be an opportunity to merge both test files under `test/python/qpy`?

Reply:
It does, but for this PR I just left it as is. I'll push a follow up PR after this merges that unifies the tests into a single module under `tests/python/qpy`